airbyte-cdk/python/airbyte_cdk/entrypoint.py

```diff
@@ -85,45 +87,65 @@ def run(self, parsed_args: argparse.Namespace) -> Iterable[str]:
                 raw_config = self.source.read_config(parsed_args.config)
                 config = self.source.configure(raw_config, temp_dir)
 
-                # Now that we have the config, we can use it to get a list of ai airbyte_secrets
-                # that we should filter in logging to avoid leaking secrets
-                config_secrets = get_secrets(source_spec.connectionSpecification, config)
-                update_secrets(config_secrets)
-
-                # Remove internal flags from config before validating so
-                # jsonschema's additionalProperties flag wont fail the validation
-                connector_config, _ = split_config(config)
-                if self.source.check_config_against_spec or cmd == "check":
-                    try:
-                        check_config_against_spec_or_exit(connector_config, source_spec)
-                    except AirbyteTracedException as traced_exc:
-                        connection_status = traced_exc.as_connection_status_message()
-                        if connection_status and cmd == "check":
-                            yield connection_status.json(exclude_unset=True)
-                            return
-                        raise traced_exc
-
                 if cmd == "check":
-                    check_result = self.source.check(self.logger, config)
-                    if check_result.status == Status.SUCCEEDED:
-                        self.logger.info("Check succeeded")
-                    else:
-                        self.logger.error("Check failed")
-
-                    output_message = AirbyteMessage(type=Type.CONNECTION_STATUS, connectionStatus=check_result).json(exclude_unset=True)
-                    yield output_message
+                    yield from map(AirbyteEntrypoint.airbyte_message_to_string, self.check(source_spec, config))
                 elif cmd == "discover":
+                    self.set_up_secret_filter(config, source_spec.connectionSpecification)
```

Comment:
nit: would it make sense to create a `discover` method to this class to stay consistent with the `check` and `read`?